WebAPI/Controllers/StudentController.cs

```diff
@@ -83,6 +85,101 @@ public async Task<IActionResult> GetStudentTimetable(string userEmail)
             return Ok(timetable);
         }
 
+        [HttpGet("getStudentTimetableCalendar/{userEmail}")]
+        public async Task<IActionResult> GetStudentTimetableCalendar(string userEmail)
+        {
+            _logger.LogInformation($"[API getStudentTimetable] Getting timetable calendar for test student");
+            User user = await _userService.GetUserByEmailAsync(userEmail);
+            string studentId = user.Student.Id.ToString();
+            bool isValidGUID = Guid.TryParse(studentId, out Guid guid);
+            if (!isValidGUID)
+            {
+                return ErrorResponse($"Student id: {studentId} is not valid GUID.");
+            }
+
+            var student = await _studentService.FindByIdAsync(guid);
+            if (student == null)
+            {
+                return ErrorResponse($"Student with id: {studentId} does not exist.");
+            }
+
+            if (student.Timetable == null)
+            {
+                return ErrorResponse($"Timetable for student with id: {studentId} does not exist.");
+            }
+            DateTime dateNow = DateTime.Now;
+            StringBuilder sb = new StringBuilder();
```

Comment:
Put the text generation into separate method of some service like CalendarGeneratorService( or similar)

Reply:
will do

Reply:
moved to calendar service